lib/pure/oids.nim

```diff
@@ -66,21 +67,23 @@ proc oidToString*(oid: Oid, str: cstring) =
   str[24] = '\0'
 
 proc `$`*(oid: Oid): string =
+  ## Converts an oid to string.
   result = newString(24)
   oidToString(oid, result)
 
-proc rand(): cint {.importc: "rand", header: "<stdlib.h>", nodecl.}
-proc srand(seed: cint) {.importc: "srand", header: "<stdlib.h>", nodecl.}
-
-var t = getTime().toUnix.int32
-srand(t)
 
 var
-  incr: int = rand()
-  fuzz: int32 = rand()
+  t = getTime().toUnix.int32
+  seed = initRand(t)
+  incr: int = seed.rand(0x7fff)
```

Comment:
this is a regression.
on OSX, `RAND_MAX = 2147483647`, so before PR, the seed has more entropy than after PR.

instead do this:
`incr: int = seed.rand(0x7fff)` => `incr: int = seed.rand(int.high)`
and further below, do this:
```nim
var i = int32(atomicInc(incr))
=>
var i = cast[int32](atomicInc(incr))
```
which also solves https://github.com/timotheecour/Nim/issues/431#issuecomment-740155796